pkg/search/postgres/common.go

```diff
@@ -110,30 +110,32 @@ func (q *query) ExtraSelectedFieldPaths() []pgsearch.SelectQueryField {
 		return nil
 	}
 
+	seenSelectPathsToIndex := make(map[string]int)
+	for idx, f := range q.SelectedFields {
+		seenSelectPathsToIndex[f.SelectPath] = idx
+	}
+
 	var out []pgsearch.SelectQueryField
 	for _, groupBy := range q.GroupBys {
-		var alreadyExists bool
-		for idx := range q.SelectedFields {
-			field := &q.SelectedFields[idx]
-			if field.SelectPath == groupBy.Field.SelectPath {
-				field.FromGroupBy = true
-				alreadyExists = true
-			}
-		}
-		if !alreadyExists {
+		idx, found := seenSelectPathsToIndex[groupBy.Field.SelectPath]
+		if !found {
 			out = append(out, groupBy.Field)
+		} else {
+			field := &q.SelectedFields[idx]
+			field.FromGroupBy = true
 		}
+
+		// -1 because the was added to `out` and not to `q.SelectedFields`.
+		seenSelectPathsToIndex[groupBy.Field.SelectPath] = -1
 	}
+
 	for _, orderByEntry := range q.Pagination.OrderBys {
-		var alreadyExists bool
-		for _, selectedField := range q.SelectedFields {
-			if selectedField.SelectPath == orderByEntry.Field.SelectPath {
-				alreadyExists = true
-			}
-		}
-		if !alreadyExists {
+		if _, found := seenSelectPathsToIndex[orderByEntry.Field.SelectPath]; !found {
 			out = append(out, orderByEntry.Field)
 		}
+
+		// -1 because the was added to `out` and not to `q.SelectedFields`.
+		seenSelectPathsToIndex[orderByEntry.Field.SelectPath] = -1
```

Comment:
Not sure if this is needed or what is desired.  If I'm following correctly the next time we look for this key in the `seenSelectPathsToIndex` map it will be found and return -1.  But we don't check for -1 in this loop and this is the last , only if it is found.  It also only matters if `OrderBys` contains duplicates.

Reply:
This is just a nice to have to make sure we are not passing duplicate order bys down to SQL engine.

Reply:
NVM.  I missed the `!` when I read this before.